src/features/events/store.ts

```diff
@@ -142,6 +142,34 @@ const eventsSlice = createSlice({
         remoteItem(participant.id, { data: participant })
       );
     },
+    participantUpdated: (
+      state,
+      action: PayloadAction<[number, ZetkinEventParticipant]>
+    ) => {
+      const [eventId, participant] = action.payload;
+      const item = state.participantsByEventId[eventId].items.find(
+        (item) => item.id === participant.id
+      );
+
+      if (item) {
+        item.data = { ...item.data, ...participant };
+        item.mutating = [];
+      } else {
+        state.participantsByEventId[eventId].items.push(
+          remoteItem(participant.id, { data: participant })
+        );
+      }
+
+      if (participant.cancelled) {
+        // If cancelled participant was contact for event, also remove contact
```

Comment:
This condition is not actually checked anywhere in the code. The contact person is just removed regardless of whether it was the participant deleted or not. So there's a bug here that needs to be fixed.

Reply:
Sorry it went a little too fast there. Fixed!